quiche/src/h3/mod.rs

```diff
@@ -931,6 +931,26 @@ pub struct Stats {
     pub qpack_decoder_stream_recv_bytes: u64,
 }
 
+fn close_conn_critical_stream(conn: &mut super::Connection) -> Result<()> {
+    conn.close(
+        true,
+        Error::ClosedCriticalStream.to_wire(),
+        b"Critical stream closed.",
+    )?;
+
+    Err(Error::ClosedCriticalStream)
+}
+
+fn check_closed_critical_stream(
```

Comment:
```suggestion
fn close_finished_critical_stream_with_error(
```

Bit of a mouthful, but seems more accurate?

Reply:
Or, idk, `close_critical_stream_if_finished(`

Reply:
I went with `close_conn_if_critical_stream_finished()`

Reply:
Oh lol, yeah, that makes way more sense.